{bio}[foss/2018b] THetA v0.7 w/ Python 2.7.15

lexming opened the pull request:
(created using `eb --new-pr`)

lexming commented:
Depends on PR #8384

lexming commented:
Test report by @lexming
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
login2.cerberus.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6126 CPU @ 2.60GHz, Python 2.7.5
See https://gist.github.com/61b4ccebb3844893ddedf18a8e618d96 for a full test report.

lexming commented:
Depends on PR #8384

Micket commented on easybuild/easyconfigs/t/THetA/THetA-0.7-GCC-8.2.0-2.31.1-Python-2.7.15.eb:
```diff
@@ -0,0 +1,39 @@
+
+dependencies = [
+    ('Python', '2.7.15'),
+    ('MATLAB', '2019a', '', True),
```
I'm a bit skeptical about these deps on both Python and MATLAB. The software seems like it would still be usable without them. It seems likely that a user could just just load whichever version of MATLAB they needed. 

Python 2.7.15 also forces python2 on any other modules that might be loaded, which severely limits the usability. I'd say this is just best left up to the user. (perhaps suggest it includes MATLAB and Python bindings in the description?)

lexming replied:
The main code is in Python and you can see that it is not Python 3 compliant. So there is no choice. 
Matlab is indeed an optional dependency (as it is Java), but I do not see any downside in having a concurrent version of Matlab loaded by default. There's no impact on performance for users not needing it, it is convenient for user using it and if any user really wants a different Matlab, it is always possible to replace the module as there are no toolchain constrains.

Micket commented on easybuild/easyconfigs/t/THetA/THetA-0.7-GCC-8.2.0-2.31.1-Python-2.7.15.eb:
```diff
@@ -0,0 +1,39 @@
+    'dirs': ['python','matlab','java/src'],
+}
+
+modextrapaths = {'PYTHONPATH': 'python'}
```
Add `'MATLABPATH': 'matlab'` ?

lexming replied:
Done :+1:

lexming commented:
Updated with a missing dependency on `bnpy` (depends on PR #8989), moved to `foss/2018b` due to that change and added comments on optional dependencies.

lexming commented:
Test report by @lexming
**FAILED**
Build succeeded for 0 out of 2 (1 easyconfigs in this PR)
login2.cerberus.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6126 CPU @ 2.60GHz, Python 2.7.5
See https://gist.github.com/51d9fa61cb88e326c1149bd69eefc212 for a full test report.

lexming commented:
PR #8989 has been merged so all checks should fine now. Any way to trigger travis again?

lexming commented:
Test report by @lexming
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
login2.cerberus.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6126 CPU @ 2.60GHz, Python 2.7.5
See https://gist.github.com/d8cae349a57bdd02b831aa1091d1875d for a full test report.

lexming commented:
PR #8989 is merged and all checks passed :heavy_check_mark:

boegel commented:
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2071.delcatty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/56801af3e28376a749411d2fe58bf1b9 for a full test report.

boegel commented:
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2705.swalot.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/7153c28710a24a63cdf5243f89070b98 for a full test report.

boegel commented:
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node3113.skitty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/4fb160b1bfaf5cf2d025535f8e7eb061 for a full test report.

boegel commented:
Going in, thanks @lexming!
